doc/sphinx/source/developer_guide/abi.rst

```diff
@@ -0,0 +1,99 @@
+.. developer_guide abi
+
+=============
+ABI messaging
+=============
+
+Paparazzi use a custom publish/subscribe middleware to exchange data between software components named ``ABI``.
+
+.. note:: This middleware was named *ABI* for *AirBorne Ivy*
+
+General Idea
+------------
+
+* Give an easy way to allow software components to exchange data with minimum delay nor execution overhead.
+* The components don't need to know each other, they only need the message format.
+* Each subscriber set a callback function that will be called when new data are sent.
+
+
+Message definition
+------------------
+
+The messages are described in ``conf/abi.xml`` analoguous to other messages in paparazzi. Name are unique, IDs starts from 0. "type" field should be a real C type. Field name might be useless.
+
+.. code-block:: xml
+
+    <class name="airborne">
+     <message name="DATA" id="0">
+      <field name="a" type="float"/>
+      <field name="b" type="struct bla"/>
+     </message>
+     ...
+    </class>
+
+Airborne code for subscriber
+----------------------------
+
+Include header and declare an ``abi_event`` as a global variable. Write the callback function with the proper prototype, matching the message.
```

Comment:
specify that `abi_event ` variables can also be `static` (but not `const`)

Reply:
Done, but is this not the default behavior in C?